### ISSUE TITLE
Is there a way to get all validation errors for required fields?

### ISSUE THREAD

**Issue by thebearingedge** [Stranger6667/jsonschema-rs#190]
Hi, I'm currently building a Web server in Rust and would like to use JSON Schema to validate client request bodies. 

I have this schema, for example:

```json
{
  "type": "object",
  "properties": {
    "user": {
      "type": "object",
      "properties": {
        "email": {
          "type": "string"
        },
        "username": {
          "type": "string"
        },
        "password": {
          "type": "string"
        }
      },
      "required": [
        "email",
        "username",
        "password"
      ]
    }
  },
  "required": [
    "user"
  ]
}
```

I am looking for a way to get all of the errors when a `"user"` is provided, but more than one required property is missing.

Input:
```json
{
  "user": {}
}
```

Current Output:
```json
[
  "'email' is a required property"
]
```

Desired Output:
```json
[
  "'email' is a required property",
  "'password' is a required property",
  "'username' is a required property"
]
```

I've played around with the `jsonschema-rs` code a little and came up with this, but I'm really new to Rust and JSON Schema in general, so it might not be a desirable implementation.

```rust
    // jsonschema/src/keywords/required.rs

    fn validate<'a, 'b>(
        &'b self,
        _: &'a JSONSchema,
        instance: &'a Value,
        instance_path: &InstancePath<'b>,
    ) -> ErrorIterator<'a> {
        if let Value::Object(item) = instance {
            let mut errors = vec![];
            for property_name in &self.required {
                if !item.contains_key(property_name) {
                    errors.push(ValidationError::required(
                        instance_path.into(),
                        instance,
                        property_name.clone(),
                    ));
                }
            }
            if errors.len() > 0 {
                return Box::new(errors.into_iter());
            }
        }
        no_error()
    }
```

I'm happy to submit a PR if you think this is on the right path or have further guidance on usage for the crate.

Thanks!

**Comment by Stranger6667** [comment]
Hi @thebearingedge 

Good catch! Your code is a perfectly valid way to fix the bug :) Maybe one improvement - `errors.len() > 0` could be replaced with `!errors.is_empty()` as `clippy` suggests + the method signature changed a little. I'd be happy to accept a PR :)

Thank you

**Comment by thebearingedge** [comment]
@Stranger6667 Great, thank you! I have opened #191 👍

**Comment by Stranger6667** [comment]
Fixed by #191 

The fix is released in `0.7.0` :tada:
